contest/contest_archive.py

```diff
@@ -38,17 +38,25 @@ def get_contests(user):
 def get_owned_or_started_contests(user):
     owned_contests = get_owned_contests(user)
     started_contests = get_started_contests()
-    return owned_contests | started_contests
+    return (owned_contests | started_contests).distinct()
+
+def get_owned_or_attended_contests(user):
+    return (get_owned_contests(user)|get_attended_contests(user)).distinct()
 
 #both owned and coowned
 def get_owned_contests(user):
     request = Q(owner = user)|Q(coowner = user)
     owned_contests = Contest.objects.order_by('-start_time').filter(request)
-    return owned_contests
+    return owned_contests.distinct()
 
 def get_started_contests():
     now = datetime.now()
-    return Contest.objects.order_by('-start_time').filter(start_time__lte = now)
+    return Contest.objects.order_by('-start_time').filter(start_time__lte = now).distinct()
+
+def get_attended_contests(user):
+    contest_list = Contestant.objects.filter(user = user).\
+                  values_list('contest', flat=True)
+    return Contest.objects.filter(pk__in=contest_list).distinct()
```

Comment:
adding that enable combining two Queryset.
I guess that the main problem is the many-to-many field(coowner)
Coowner's data is stored in many rows, since MySQL DB only can store one data per field.
So one Contest may be conbined by several rows.
When querying, Models return those rows, and duplicate problem occurs.
I think that's the reason why we should use `distinct()`
我的觀察啦